Added VPN and Proxmox doc (#137)
hyphacoop/handbook: guides-infrastructure-advanced.md

### Contents
- [Proxmox](#proxmox)
    - [Creating Cloud-init templates](#creating-cloud-init-templates)
    - [Manually creating Virtual Machines](#manually-creating-virtual-machines)
- [OVH Hosting](#ovh-hosting)
    - [Setting up OVH backup space](#setting-up-ovh-backup-space)
- [Backup and Restore](#backup-and-restore)
    - [Proxmox VMs](#proxmox-vms)
    - [FreeNAS Dynamic Content](#freenas-dynamic-content)
    - [Passbolt Backup](#passbolt-backup)

## Proxmox

### Creating Cloud-init templates

Cloud-init templates are used for easy deployment of VMs. Cloud-init allows auto configuration such as auto resize filesystem, network settings, default username and its SSH keys.
This only needs to be done once per distro.

These tasks need to be done over SSH tunnel or over the VPN

- Debian 10
  ```

>   wget https://cdimage.debian.org/cdimage/openstack/10.4.3-20200610/debian-10.4.3-20200610-openstack-amd64.qcow2

Reviewer: ```suggestion
  wget https://cdimage.debian.org/cdimage/openstack/10.11.0/debian-10.11.0-openstack-amd64.qcow2 -O debian-10.qcow2
```
Images changed

  qm create 9000 --memory 1024 --net0 virtio,bridge=vmbr2 --name cloud-init-debian-10

>   qm importdisk 9000 debian-10.4.3-20200610-openstack-amd64.qcow2 local --format qcow2

Reviewer: ```suggestion
  qm importdisk 9000 debian-10.qcow2 local --format qcow2
```

  qm set 9000 --scsihw virtio-scsi-pci --virtio0 local:9000/vm-9000-disk-0.qcow2
  qm set 9000 --cpu host
  qm set 9000 --ide2 local:cloudinit
  qm set 9000 --boot c --bootdisk virtio0
  qm set 9000 --serial0 socket --vga serial0
  qm set 9000 --vga std
  qm set 9000 --ostype l26
  ```

>   We need to install `resolvconf` to update resolv.conf from Cloud-init
>   Attach a CD-ROM and select Debian live ISO
>   Make sure boot order is set to CD ROM first
>   Start the VM
>   After booted we need to install `resolvconf`
>   Follow the following commands:

Reviewer: ```suggestion
  We need to modify the stock image to include  `resolvconf` to update `resolv.conf` from Cloud-init
  We will do this by mapping the qcow2 disk image to a `nbd` device, mounting it, and making the modifications in a `chroot` environment
    
```

  ```

>   sudo -s
>   mount /dev/vda1 /mnt
>   chroot /mnt
>   echo "nameserver 1.1.1.1" > /etc/resolv.conf
>   apt update
>   apt install resolvconf
>   exit
>   umount /mnt
>   shutdown -h now

Reviewer: ```suggestion
modprobe nbd max_part=8
qemu-nbd --connect=/dev/nbd0 /var/lib/vz/images/9000/vm-9000-disk-0.qcow2
mkdir tmp
mount mount /dev/nbd0p1 tmp
cd tmp
chroot .
echo "nameserver 1.1.1.1" > /etc/resolv.conf
apt update
apt install resolvconf
exit
cd ..
umount tmp
rm -rf tmp
qemu-nbd --disconnect /dev/nbd0
sleep 5
rmmod nbd
```

  ```

>   Remove CD-ROM device and set boot order back to `virtio0`

Reviewer: ```suggestion
```

#### Create a VM using Proxmox

6. To create the VM go login to KVM1 Proxmox host <a href="https://kvm1.hypha.coop:8006/" data-proofer-ignore>here</a> using username root with the password in Passbolt [here](https://pass.hypha.coop/app/passwords/view/a34731d5-eb6a-4f0c-9475-7839280e529b)
7. Select a template to clone in the `Datacenter` sidebar under `9XXX`
8. Click on `More` on the top right corner and select `Clone`
9. `VMID` is the VMID you noted above
10. Mode must be `Full Clone` 
11. Choose Target Storage depending where you want the VM to be running from.
    - local: SSD
    - local-hdds: Spinning drives
12. Name should be the public FQDN or *.hypha.prod/stg domain for local access only VMs. 
13. Format should be qcow2 (default)
14. Press clone, once finished we can set hardware 
15. Select the VM ID in the `Datacenter` sidebar and click on `Hardware`
16. Modify the memory, number of processors and increase diskspace as needed.
    - Recommended numbers
        - memory: 1024MB

>         - CPU: 1 socket, 4 cores

Reviewer: Most servers only need 1 or 2 cores. I would lower this to prevent too high overprovisioning
```suggestion
        - CPU: 1 socket, 2 cores
```
